tooling/cli/lib/helpers/atoms.js

```diff
@@ -0,0 +1,70 @@
+const { getFilePaths } = require('./filepaths');
+const path = require('path');
+const { access, readFile, writeFile } = require('fs/promises');
+
+const dirHasAtoms = async dir => {
+  const contents = await getFilePaths(dir);
+  const hasAtomsDir = contents.some(name => /\/atoms\//.test(name));
+  const hasAtomsConfig = contents.some(name => /\/vite.atoms.config.(ts|tsx|js|jsx)$/.test(name));
+
+  return hasAtomsDir && hasAtomsConfig;
+};
+
+const getAtomsHash = async (dir, type) => {
+  const hashedAssets = (await getFilePaths(path.join(dir, '/dist/assets'))) || [];
+  if (!hashedAssets.length) {
+    throw new Error('No compiled Atoms assets found.');
+  }
+
+  const regex =
+    type === 'script' ? /\/?index-([a-zA-Z]|\d)+.(ts|tsx|js|jsx)$/ : /\/?style-([a-zA-Z]|\d)+.css$/;
+
+  const idx = hashedAssets.find(asset => regex.test(asset));
+
+  if (!idx && type !== 'style') {
+    throw new Error('No Atoms index found.');
+  }
+
+  const idxHash = idx.match(/-(([a-zA-Z]|\d)+\.)/)[0].replace(/(-|\.)/g, '');
```

Comment:
`idx` could be undefined here when `type = 'style'` and there is no style file found, we may want to just return early ?